tests/AttributeReleasePolicyTest.php

```diff
@@ -72,4 +72,69 @@ public function testInvalidInstantiation()
         }
         $this->assertNotNull($e);
     }
+
+    public function testArpWithSources()
+    {
+        $policy = new AttributeReleasePolicy(
+            array(
+                'a' => array('a'),
+                'b' => array(
+                    array(
+                        'value' => 'b',
+                        'source' => 'b',
+                    ),
+                ),
+            )
+        );
+
+        $this->assertEquals(array('a', 'b'), $policy->getAttributeNames());
+        $this->assertTrue($policy->hasAttribute('a'));
+        $this->assertTrue($policy->hasAttribute('b'));
+        $this->assertTrue($policy->isAllowed('a', 'a'));
+        $this->assertTrue($policy->isAllowed('b', 'b'));
+        $this->assertFalse($policy->isAllowed('a', 'b'));
+        $this->assertFalse($policy->isAllowed('b', 'a'));
+    }
+
+    public function testInvalidArpWithSourceSpecification()
+    {
+        $this->setExpectedException(InvalidArgumentException::class);
+
+        $policy = new AttributeReleasePolicy(
```

Comment:
Personally I would not use a declaration here and just instantiate a `new AttributeReleasePolicy` into 'void'.